dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -571,12 +571,14 @@ public ListenableFuture<List<RouteFailure>> copy(List<Route> routes, UfsReadOpti
     List<RouteFailure> errors = Collections.synchronizedList(new ArrayList<>());
     FileSystemContext fsContext = FileSystemContext.create(Configuration.global());
     for (Route route : routes) {
-      FileSystem srcFs = FileSystem.Factory.create(fsContext,
-          FileSystemOptions.create(Configuration.global(),
-              Optional.of(new UfsFileSystemOptions(new AlluxioURI(route.getSrc()).getRootPath()))));
-      FileSystem dstFs = FileSystem.Factory.create(fsContext,
-          FileSystemOptions.create(Configuration.global(),
-              Optional.of(new UfsFileSystemOptions(new AlluxioURI(route.getDst()).getRootPath()))));
+      UnderFileSystem srcUfs = mUfsManager.getOrAdd(new AlluxioURI(route.getSrc()),
+          UnderFileSystemConfiguration.defaults(mConf));
+      FileSystem srcFs = new UfsBaseFileSystem(fsContext,
```

Comment:
Do we have the logics to close the created UfsBaseFileSystem?

Reply:
Fixed, but I think the current Dora ufs manager doesn't really support multiple mounts (ufs path). We should revisit this part after we sort out how we can support the UFS path credentials in 3.x